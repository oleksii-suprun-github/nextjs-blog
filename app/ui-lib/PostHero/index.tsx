import Image from 'next/image';
import Breadcrumbs from '@/app/components/Breadcrumbs';

type PostHeroType = {
  creationDate: string;
  title: string;
  introduction: string;
  image: {
    alt?: string;
    url: string;
  };
};

function PostHero({ creationDate, title, introduction, image }: PostHeroType) {
  return (
    <>
      <header className="px-4">
        <div className="container mx-auto">
          <div className="mb-16">
            <Breadcrumbs />
          </div>
          <p className="mb-5">
            <em>Published {creationDate}</em>
          </p>
          <h1 className="mb-10 max-w-2xl text-4xl font-bold leading-normal lg:text-5xl lg:leading-normal">
            {title}
          </h1>
          <p className="mb-24 max-w-lg text-xl leading-normal lg:mb-32 lg:leading-relaxed">
            {introduction}
          </p>
        </div>
      </header>
      <figure className="absolute left-0 top-0 -z-10 h-[150vh] w-full overflow-hidden after:absolute after:inset-0 after:h-full after:bg-brand-dark-purple after:bg-opacity-70 after:content-['']">
        <Image
          src={image.url}
          alt={image.alt || title}
          width={1920}
          height={1080}
          quality={60}
          priority
          className="h-full w-full object-cover lg:h-auto lg:w-screen"
          placeholder="blur"
          blurDataURL="iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkmHOiHgAD7QHlxT90/wAAAABJRU5ErkJggg=="
          sizes="(max-width: 768px) 100vw, 400px"
        />
      </figure>
    </>
  );
}

export default PostHero;
