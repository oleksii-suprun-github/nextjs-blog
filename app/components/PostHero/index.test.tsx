import { render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import PostHero from '.';

// Mock next/image
vi.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    const { src, alt, ...rest } = props;
    return <img src={src} alt={alt} {...rest} />;
  },
}));

// Mock Breadcrumbs
vi.mock('@/app/components/Breadcrumbs', () => ({
  __esModule: true,
  default: () => <nav aria-label="breadcrumb">Breadcrumbs</nav>,
}));

describe('PostHero Component', () => {
  const mockProps = {
    creationDate: 'April 27, 2024',
    title: 'Understanding React Testing',
    introduction:
      'This post delves into the best practices for testing React components using modern tools.',
    image: {
      alt: 'mock image',
      url: '/images/mock-image.png',
    },
  };

  it('renders correctly and matches the snapshot', () => {
    const { asFragment } = render(<PostHero {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
