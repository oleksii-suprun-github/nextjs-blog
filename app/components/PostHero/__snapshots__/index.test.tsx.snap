// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostHero Component > renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="px-4"
  >
    <div
      class="container"
    >
      <div
        class="mb-16"
      >
        <nav
          aria-label="breadcrumb"
        >
          Breadcrumbs
        </nav>
      </div>
      <p
        class="mb-5"
      >
        <em>
          Published April 27, 2024
        </em>
      </p>
      <h1
        class="mb-10 max-w-2xl text-4xl font-bold leading-normal lg:text-5xl lg:leading-normal"
      >
        Understanding React Testing
      </h1>
      <p
        class="mb-24 max-w-lg text-xl leading-normal lg:mb-32 lg:leading-relaxed"
      >
        This post delves into the best practices for testing React components using modern tools.
      </p>
    </div>
  </header>
  <figure
    class="absolute left-0 top-0 -z-10 h-[150vh] w-full overflow-hidden after:absolute after:inset-0 after:h-full after:bg-brand-dark-purple after:bg-opacity-85 after:content-['']"
  >
    <img
      alt="mock image"
      blurdataurl="iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkmHOiHgAD7QHlxT90/wAAAABJRU5ErkJggg=="
      class="h-full w-full object-cover lg:h-auto lg:w-screen"
      height="1080"
      placeholder="blur"
      quality="60"
      sizes="(max-width: 768px) 100vw, 400px"
      src="/images/mock-image.png"
      width="1920"
    />
  </figure>
</DocumentFragment>
`;
